src/core/style/image_style_change.cpp

```diff
@@ -0,0 +1,31 @@
+// Copyright (C) 2023 MapLibre contributors
+
+// SPDX-License-Identifier: BSD-2-Clause
+
+#include "image_style_change_p.hpp"
+
+#include <QMapLibre/Map>
+
+namespace QMapLibre {
+
+// StyleAddImage
+StyleAddImage::StyleAddImage(const StyleParameter *parameter)
+    : m_id(parameter->property("id").toString()),
+      m_sprite(QImage(parameter->property("sprite").toString())) {}
+
+void StyleAddImage::apply(Map *map) {
```

Comment:
Nit: Would be nice to have null deref check here since it's being passed a raw map pointer (same below in `StyleRemoveImage`

Reply:
Done.